iModelCore/ECDb/Tests/NonPublished/SchemaUpgradeTests.cpp

```diff
@@ -10714,63 +10714,126 @@ TEST_F(SchemaUpgradeTestFixture, RemoveKindOfQuantityFromECPropertyUsingCA)
+            <ECSchemaReference name="Units" version="01.00.00" alias="u" />
+            <ECSchemaReference name="Formats" version="01.00.00" alias="f" />
+        </ECSchema>)schema");
 
     auto options = SchemaManager::SchemaImportOptions::DoNotFailForDeletionsOrModifications;
 
     ASSERT_EQ(SUCCESS, GetHelper().ImportSchema(modifiedSchemaItem, options))
         << "Illegal KoQ modification should not fail when DoNotFail flag is on";   
 
-    ECSchemaCP schema = m_ecdb.Schemas().GetSchema("Schema1");
-    KindOfQuantityCP koq = (*schema).GetKindOfQuantityCP("K3");
-    ASSERT_TRUE(koq != nullptr) << "KindOfQuantity K3 should still exist";
+    ECSchemaCP schema = m_ecdb.Schemas().GetSchema("TestSchema");
+    KindOfQuantityCP koq = (*schema).GetKindOfQuantityCP("KindOfQuantity1");
+    ASSERT_TRUE(koq != nullptr) << "KindOfQuantity 'KindOfQuantity1' should still exist";
     }
 
 //---------------------------------------------------------------------------------------
 // @bsimethod
 //+---------------+---------------+---------------+---------------+---------------+------
 TEST_F(SchemaUpgradeTestFixture, KoQModificationWithDoNotFailFlag)
     {
-    ASSERT_EQ(SUCCESS, SetupECDb("schemaupgrade_KindOfQuantity.ecdb", SchemaItem(R"xml(<?xml version="1.0" encoding="utf-8" ?>
-                            <ECSchema schemaName="Schema1" alias="s1" version="1.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">
-                                    <KindOfQuantity typeName="K1" description="My KOQ 1" displayLabel="KOQ 1" persistenceUnit="CM" relativeError="1" />
-                                    <KindOfQuantity typeName="K2" description="My KOQ 2" displayLabel="KOQ 2" persistenceUnit="M" presentationUnits="FT;IN" relativeError="2" />
-                                    <KindOfQuantity typeName="K3" description="My KOQ 3" displayLabel="KOQ 3" persistenceUnit="KG" presentationUnits="G" relativeError="3" />
-                                    <KindOfQuantity typeName="K4" description="My KOQ 4" displayLabel="KOQ 4" persistenceUnit="G" presentationUnits="MG" relativeError="4" />
-                            </ECSchema>)xml")));
+    ASSERT_EQ(SUCCESS, SetupECDb("schemaupgrade_KindOfQuantity.ecdb", SchemaItem(R"schema(<?xml version='1.0' encoding='utf-8'?>
+        <ECSchema schemaName='TestSchema' alias='ts' version='1.0.0' xmlns='http://www.bentley.com/schemas/Bentley.ECXML.3.2'>
+            <ECSchemaReference name="Units" version="01.00.00" alias="u" />
+            <ECSchemaReference name="Formats" version="01.00.00" alias="f" />
+            <KindOfQuantity 
+                typeName='KindOfQuantity1' 
+                description='KindOfQuantity1'
+                displayLabel='KindOfQuantity1'
+                persistenceUnit='u:CM'
+                relativeError='.5'
+                presentationUnits='f:AmerFI[u:FT][u:IN]' 
+            />

```

Comment:
You can slim down these definitions by skipping description and presentationUnits.  I would also go back to shorter names and display labels.  Once you have done that you can make the KOQ definitions a single line which will keep the tests shorter and easier to read.  Consider putting the persistenceUnit as the last attribute because that is the attribute you are changing.

Also I try to keep the quoting consistent within one string.  In these schemas you're switching between single and double quotes ... I suggest double quotes.